src/annotator/components/Toolbar.tsx

```diff
@@ -146,8 +146,22 @@ export default function Toolbar({
   return (
     <div
       className={classnames(
-        'absolute left-[-33px] w-[33px] z-2',
+        {
+          // For minimal controls, display the toolbar to the left, fully
+          // outside the sidebar
+          'absolute right-full': useMinimalControls,
+          // When the full toolbar is displayed, we position it relative to its
+          // container, so that it takes vertical space and pushes next elements
+          // down (eg. the buckets list).
+          // We push it away from the right side to compensate for the parent
+          // hardcoded width.
+          'relative right-[11px]': !useMinimalControls,
+        },
+        'w-[33px] z-2',
         'text-px-base leading-none', // non-scaling sizing
+        // Allow pointer events in the toolbar, in case they were disabled in
+        // the parent container
+        'pointer-events-auto',
```

Comment:
This could be moved to the `toolbarContainer` element's styles in `sidebar.tsx` to bring them closer to where `pointer-events: none` is applied.